underlay/json/defaultpreferences.json

```diff
@@ -0,0 +1,5 @@
+{
+  "autoExpandPools": false,
```

Comment:
I reckon this should probably be generated by the script using the xml config instead. Ideally we should not spread out the config settings if we can avoid it. What do you think?

Reply:
Yes, the default preferences should probably all come from the xml, and either all be in the index.html, or all be written to defaultpreferences.json
We're almost to the point where index.html is a static file. If preferences and the "build date" info were also in preferences json, that might be all that's left?
definedRules is not used anymore, I'll remove that one.
Thoughts?